adit/dicom_web/utils/wadors_utils.py

```diff
@@ -31,6 +35,15 @@ async def wado_retrieve(
     queue = asyncio.Queue[Dataset]()
 
     def callback(ds: Dataset) -> None:
+        if pseudonym is not None:
+            anonymizer = Anonymizer()
```

Comment:
Nope. Not possible this way. It would create a new Anonymizer on every image. Do you remember what we said about retaining the state of the anonymizer regarding a whole study? Your code would disrupt the whole study.

We also discussed to refactor the Anonymizer into our own `Pseudonymizer` class and use that here and in the processor